activesupport/lib/active_support/callbacks.rb

```diff
@@ -400,21 +401,17 @@ def initialize(target, method, arguments, block)
         # The actual invocation is left up to the caller to minimize
         # call stack pollution.
         def expand(target, value, block)
-          result = @arguments.map { |arg|
-            case arg
-            when :value; value
-            when :target; target
-            when :block; block || raise(ArgumentError)
-            end
-          }
+          expanded = [@override_target || target, @override_block || block, @method_name]
 
-          result.unshift @method_name
-          result.unshift @override_block || block
-          result.unshift @override_target || target
+          @arguments.each do |arg|
+            expanded << case arg
```

Comment:
Do you see @kaspth's suggestion?

https://github.com/rails/rails/pull/37969#discussion_r358354090

> I'm not liking the specific style here, any way we can do something like this?

Reply:
Yeah, sorry I should have left a comment. I pushed my branch, but I'm still exploring other styles.